option.go

```diff
@@ -287,6 +287,13 @@ func OptionHide() func(*BaseRoute) {
 	}
 }
 
+// Hide hides the route from the OpenAPI spec.
+func OptionDefaultStatusCode(defaultStatusCode int) func(*BaseRoute) {
+	return func(r *BaseRoute) {
+		r.DefaultStatusCode = defaultStatusCode
+	}
+}
```

Comment:
I don't get why you keep adding new things at the root, people should use the ones in options.

Is it a dependency cycle issue?

Reply:
Yes it is

Reply:
I think you could sort this by adding a new package like `route` that will contain only BaseRoute. You may have issues with OpenAPIParam at some point...

Anyway dependency cycle is never fun, so I can get you don't want to invest on fixing them.

Anyway, I think you should add the `// Deprecated: use option.DefaultStatusCode` here and for all existing code.

It should be OK, I don't think IDE will report the fact option.DefaultStatusCode is deprecated because it uses OptionDefaultStatusCode

Reply:
Yes why not